homeassistant/components/keenetic_ndms2/config_flow.py

```diff
@@ -0,0 +1,146 @@
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+
+from .const import (
+    CONF_CONSIDER_HOME,
+    CONF_INCLUDE_ARP,
+    CONF_INCLUDE_ASSOCIATED,
+    CONF_INTERFACES,
+    CONF_TRY_HOTSPOT,
+    DEFAULT_CONSIDER_HOME,
+    DEFAULT_INTERFACE,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_TELNET_PORT,
+    DOMAIN,
+)
+
+
+class KeeneticFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return KeeneticOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            for entry in self.hass.config_entries.async_entries(DOMAIN):
+                if entry.data[CONF_HOST] == user_input[CONF_HOST]:
+                    return self.async_abort(reason="already_configured")
+
+            _client = Client(
+                TelnetConnection(
+                    user_input[CONF_HOST],
+                    user_input[CONF_PORT],
+                    user_input[CONF_USERNAME],
+                    user_input[CONF_PASSWORD],
+                    timeout=10,
+                )
+            )
+
+            try:
+                router_info = await self.hass.async_add_executor_job(
+                    _client.get_router_info
+                )
+            except ConnectionException:
+                errors["base"] = "cannot_connect"
+            else:
+                return self.async_create_entry(title=router_info.name, data=user_input)
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST): str,
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                    vol.Optional(CONF_PORT, default=DEFAULT_TELNET_PORT): int,
+                }
+            ),
+            errors=errors,
+        )
+
+
+class KeeneticOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle options."""
+
+    def __init__(self, config_entry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, _user_input=None):
+        """Manage the options."""
+        return await self.async_step_device_tracker()
+
+    async def async_step_device_tracker(self, user_input=None):
+        """Manage the device tracker options."""
+        if user_input is not None:
+            return self.async_create_entry(
+                title="",
+                data={
+                    **user_input,
+                    CONF_INTERFACES: [
```

Comment:
I'm worried about this option since it's very easy for the user to set an incorrect option and the only feedback to the user for this is the lack of device updates.

Maybe we should not include this option until we can some validation?

Reply:
I understand your concerns but this configuration was available before (single interface selection though).

I've done some trials of getting the list of interface and it needs some thinking thru both parsing it and filtration since the list contains all the low level interfaces which are not feasible for this configuration.

Now when you say "validation" I think I have an idea: even without getting the list it's possible to validate values and spot interfaces that are not available

Reply:
Invested some time into the library module development and I have a list of interfaces now.

Reply:
How does it looks weird?

Reply:
Like this:
![Снимок экрана от 2021-02-12 15-57-11](https://user-images.githubusercontent.com/1551926/107748035-00e64d80-6d4b-11eb-959a-e0a4e6ed3ab0.png)

Looks a bit uglier than a conventional select control IMO

Reply:
Ok. I'm not a frontend connoisseur. 😃